lib/report.js

```diff
@@ -158,7 +160,144 @@ class ConciseReporter extends Reporter {
+  }
+
+  error(test, error) {
+    // remove one success because 'error' may only be received after end
+    // though the test may have already failed so we need a way to check
+    // if we should do this, i.e. store map of successful tests
+    // this.successes--;
+    // test.todo ? this.skipped++ : this.failures++;
+    // TODO(lundibundi): add test.metadata.filename here too once available
+    this.listFailure(
+      test,
+      { message: `Failure after end: ${error ? error.message : 'no message'}` },
+      ` Test failed: ${test.caption}`
+    );
+  }
+
+  parseTestResults(test, offset = 0) {
+    if (!test.success) {
+      if (!test.todo) {
+        this.failedTests.push(this.testNumeration);
+      }
+      this.listFailure(test, true);
+      this.got.forEach(el => {
+        this.message += `${pad(6)}- ${el}\n`;
+      });
+      this.message += `${pad(4)}expect: \n`;
+      this.expected.forEach(el => {
+        this.message += `${pad(6)}- ${el}\n`;
+      });
+      this.message += '  ...';
+      this.got = [];
+      this.expected = [];
+      console.log(this.message);
+    } else {
+      if (offset === 0) this.testNumeration++;
+      test.results.forEach((res, index) => {
+        const testId = index + 1;
+        if (res.success) {
+          this.message = pad(offset + 2);
+          if (res.type === 'subtest') {
+            const t = res.test;
+            console.log(`${this.message}# Subtest: ${t.caption}`);
+            this.parseTestResults(t, offset + 2);
+            const prefix = t.success ? 'ok' : 'not ok';
+            this.message = pad(offset + 2);
+            let message = `${this.message}${prefix} ${testId}`;
+            message += ` - ${t.caption}`;
+            console.log(message);
+          } else {
+            this.checkSuccess(res, testId);
+          }
+        }
+      });
+      if (test.results.length === 0) this.message = pad(offset + 2);
+      console.log(`${this.message}1..${test.results.length}`);
+    }
+  }
+
+  checkSuccess(test, id) {
+    const testPrefix = test.success ? 'ok' : 'not ok';
+    const successfulTest = `${this.message}${testPrefix} ${id}`;
+    console.log(successfulTest);
+  }
+
+  listFailure(test, firstFailure) {
+    test.results.forEach(res => {
+      if (!res.success && firstFailure) {
+        firstFailure = false;
+        this.message += '  ---\n';
+        if (res.message) this.message += `  message: '${res.message}'\n`;
+        this.message += '  severity: ';
+        this.message += res.severity ? `${res.severity}\n` : 'fail\n';
+        this.message += `${pad(2)}data:\n`;
+        this.message += `${pad(4)}got:\n`;
+      }
+      const [expected, actual] = [res.expected, res.actual].map(res =>
+        typeof res === 'function' || res instanceof Error ? res.toString() : res
+      );
+      if (res.test) {
+        this.listFailure(res.test);
+      } else {
+        this.got.push(actual);
+        this.expected.push(expected);
```

Comment:
Why are all actual/expected merged into an array? 
Each assertion (record in `results` array) is a test on its own that should be printed with ok/not ok.